Bindings/Java/Matlab/examples/Moco/example2DWalking/example2DWalkingStepAsymmetry.m

```diff
@@ -0,0 +1,472 @@
+%
+% Users must also prescribed the stride length via 'setStrideLength()'. The goal 
+% then calculates the minimum and maximum bounds on the distance between right 
+% and left foot. Users must ensure that this stride length is met via problem
+% bounds or other goals; the value provided to MocoStepLengthAsymmetryGoal is 
+% only used to compute the model's asymmetry in the cost function.
+%
+% Because this goal doesn't directly compute the step length asymmetry from
+% heel strike data, users should confirm that the step length asymmetry
+% from the solution matches closely to their target. To do this, we
+% provide the helper function computeStepAsymmetryValues() below. Users may
+% also want to confirm that the stride length from the optimization
+% matches with setStrideLength(), or set additional constraints for stride length
+% within the optimization. Additionally, in some cases users may want to set 
+% target asymmetries above or below the desired value, in the event there is 
+% some offset.
+function stepLengthAsymmetry()
+
+import org.opensim.modeling.*;
+
+% Create a MocoStudy
+% ------------------
+study = MocoStudy();
+study.setName('example2DWalking_StepLengthAsymmetry');
+
+% Define the MocoProblem
+% ----------------------
+problem = study.updProblem();
+modelProcessor = ModelProcessor('2D_gait.osim');                   
+modelProcessor.append(ModOpTendonComplianceDynamicsModeDGF('implicit'));
+problem.setModelProcessor(modelProcessor);   
+problem.setTimeBounds(0, 0.94);
+
+% Goals
+% =====
+
+% Periodicity 
+periodicityGoal = MocoPeriodicityGoal('periodicity');
+model = modelProcessor.process();
+model.initSystem();
+% All states are periodic except for the value of the pelvis_tx coordinate.
+for i = 1:model.getNumStateVariables()
+   currentStateName = string(model.getStateVariableNames().getitem(i-1));
+  if ~contains(currentStateName,'pelvis_tx/value')
+     periodicityGoal.addStatePair(MocoPeriodicityGoalPair(currentStateName));
+  end
+end
+% The lumbar actuator control is periodic.
+periodicityGoal.addControlPair(MocoPeriodicityGoalPair('/lumbarAct'));
+problem.addGoal(periodicityGoal);
+
+% Average gait speed
+% ------------------
+speedGoal = MocoAverageSpeedGoal('speed');
+speedGoal.set_desired_average_speed(1.0);
+problem.addGoal(speedGoal);
+
+% Effort over distance
+% --------------------
+effortGoal = MocoControlGoal('effort', 10.0);
+effortGoal.setExponent(3);
+effortGoal.setDivideByDisplacement(true);
+problem.addGoal(effortGoal);
+
+% Step length asymmetry
+% ---------------------
+stepLengthAsymmetry = MocoStepLengthAsymmetryGoal();
+stepLengthAsymmetry.setWeight(5);
+% Provide the body name for the right foot.
+stepLengthAsymmetry.setRightFootFrame('/bodyset/calcn_r');    
+% Provide the body name for the left foot.
+stepLengthAsymmetry.setLeftFootFrame('/bodyset/calcn_l');     
+% Value for smoothing term use to compute asymmetry (default is 500). Users may 
```

Comment:
Default is 5 now, not 500. Maybe add note about how the smoothing is scaled *100 in the goal itself.